test/remote/gateways/remote_airwallex_test.rb

```diff
@@ -0,0 +1,140 @@
+require 'test_helper'
+
+class RemoteAirwallexTest < Test::Unit::TestCase
+  def setup
+    @gateway = AirwallexGateway.new(fixtures(:airwallex))
+
+    # https://www.airwallex.com/docs/online-payments__test-card-numbers
+    @amount = 100
+    @declined_amount = 8014
+    @credit_card = credit_card('4000100011112224')
+    @declined_card = credit_card('2223 0000 1018 1375')
+    @options = { return_url: 'https://example.com' }
+  end
+
+  def test_successful_purchase
+    response = @gateway.purchase(@amount, @credit_card, @options)
+    assert_success response
+    assert_equal 'AUTHORIZED', response.message
+  end
+
+  def test_successful_purchase_with_more_options
+    response = @gateway.purchase(@amount, @credit_card, @options.merge(address))
+    assert_success response
+    assert_equal 'AUTHORIZED', response.message
+  end
+
+  def test_failed_purchase
+    response = @gateway.purchase(@declined_amount, @declined_card, @options)
+    assert_failure response
+    assert_equal 'The card issuer declined this transaction. Please refer to the original response code.', response.message
+    assert_equal '14', response.error_code
+  end
+
+  def test_purchase_with_insufficient_data_returns_useful_message
+    options = {}.merge(@options)
+    options.delete(:return_url)
+    response = @gateway.purchase(@amount, @credit_card, options)
+    assert_failure response
+    assert_equal "Missing 'return_url' for card 3DS verification.", response.message
+    assert_equal 'validation_error', response.error_code
+  end
+
+  def test_successful_authorize_and_capture
+    auth = @gateway.authorize(@amount, @credit_card, @options)
+    assert_success auth
+
+    assert capture = @gateway.capture(@amount, auth.authorization, @options)
+    assert_success capture
+    assert_equal 'CAPTURE_REQUESTED', capture.message
+  end
+
+  def test_failed_authorize
+    response = @gateway.authorize(@declined_amount, @declined_card, @options)
+    assert_failure response
+    assert_equal 'The card issuer declined this transaction. Please refer to the original response code.', response.message
+  end
+
+  def test_partial_capture
+    auth = @gateway.authorize(@amount, @credit_card, @options)
+    assert_success auth
+
+    assert capture = @gateway.capture(@amount - 1, auth.authorization, @options)
+    assert_success capture
+  end
+
+  def test_failed_capture
+    response = @gateway.capture(@declined_amount, '12345', @options)
+    assert_failure response
+    assert_match(/The requested endpoint does not exist/, response.message)
```

Comment:
🤔  This seems like a weird message to return

Reply:
oh right, that makes more sense. Because the ID is in the url the endpoint literally doesn't exist.

Seems like a minor wart in the gateway API considering refund has a nicely-tailored message.